cpp/ps0005/Task5.cpp

```diff
@@ -17,59 +18,62 @@ int main()
-                }
-                else if (x == 0 && y == 0)
-                {
-                    cout << "Infinite number of solution= (0/0) undefined quantity. \n";
-                }
-                else
-                {
-                     z= x/y;
-                     cout << "(" << x << ") / "  "(" << y <<") =" << z <<"\n";
-                }
-                break;
+    // checking user inputs and processing 4/4
+    // printing the result if all data is valid 1/1
+    switch (r)
+    {
+        case  '/' :
+            if(x != 0 && y == 0 )
+            {
+                cout << "Math error: cannot divide by 0 \n";
+            }
+            else if (x == 0 && y == 0)
+            {
+                cout << "Infinite number of solution= (0/0) undefined quantity. \n";
+            }
+            else
+            {
+                 z= x/y;
+                 cout << "(" << x << ") / "  "(" << y <<") =" << z <<"\n";
+            }
+            break;
 
-            case '*':
-                 z= x*y;
-                 cout << "(" << x << ") * "  "(" << y <<") =" << z <<"\n";
-                 break;
+        case '*':
+             z= x*y;
+             cout << "(" << x << ") * "  "(" << y <<") =" << z <<"\n";
+             break;
 
-            case '+':
-                 z= x+y;
-                 cout << "(" << x << ") + "  "(" << y <<") =" << z <<"\n";
-                 break;
+        case '+':
+             z= x+y;
+             cout << "(" << x << ") + "  "(" << y <<") =" << z <<"\n";
+             break;
 
-            case '-':
-                 z= x-y;
-                 cout << "(" << x << ") - "  "(" << y <<") =" << z <<"\n";
-                 break;
+        case '-':
+             z= x-y;
+             cout << "(" << x << ") - "  "(" << y <<") =" << z <<"\n";
+             break;
 
 
-            case '%':
-                 xx= (int) x;
-                 yy= (int) y;
+        case '%':
+             xx= (int) x;
+             yy= (int) y;
 
-                 if (xx == x && yy == y)
-                 {
-                     z= xx % yy;
-                 cout << "(" << xx << ") % "  "(" << y <<") =" << z <<"\n";
-                 }
-                 else
-                 {
-                     cout << "cannot execute modulo operation for floating point numbers.";
-                 }
+             if (xx == x && yy == y)
+             {
+                 z= xx % yy;
+             cout << "(" << xx << ") % "  "(" << y <<") =" << z <<"\n";
+             }
+             else
+             {
+                 cout << "cannot execute modulo operation for floating point numbers.";
+             }
 
-                 break;
+             break;
 
-             default:
-                cout <<"Invalid character in operator \n";
-                break;
-        }
+         default:
+            cout <<"Invalid character in operator \n";
+            break;
+    }

```

Comment:
- [x] used `switch` and `if` structures in a right way to handle different cases
- [x] handled division operation `/`
- [x] handled modulo operation `%`
- [x] handled invalid operator case
- [x] chose the right operation for each case
- [x] printed the result of the chosen operation